cupy/linalg/einsum.py

```diff
@@ -196,13 +198,174 @@ def calc_combined_view(ioperands, subscripts):
     return result.reshape(a_shape_stack + b_shape_stack), subscript
 
 
-def get_dummy_labels(label_list):
-    dummy_label_set = set()
-    count_dict = collections.Counter(label_list)
-    for label, count in six.iteritems(count_dict):
-        if label != '@' and count >= 2:
-            dummy_label_set.add(label)
-    return dummy_label_set
+def parse_einsum_input(operands):
```

Comment:
This was because private functions in this file (e.g., `calc_single_view`) are named using that convention.
But I agree that it shouldn't be public. I renamed it to `_parse_einsum_input`.